junit-platform-commons/src/main/java/org/junit/platform/commons/util/ReflectionUtils.java

```diff
@@ -840,6 +842,25 @@ private static void findNestedClasses(Class<?> clazz, Set<Class<?>> candidates)
 	}
 
 	/**
+	 * Method comparator based upon JUnit4 org.junit.internal.MethodSorter implementation.
+	 */
+	private static int defaultMethodSorter(Method method1, Method method2) {
+		if (method1 == method2) {
+			return 0;
+		}
+		String name1 = method1.getName();
+		String name2 = method2.getName();
+		int comparison = Integer.compare(name1.hashCode(), name2.hashCode());
+		if (comparison == 0) {
+			comparison = name1.compareTo(name2);
+			if (comparison == 0) {
+				comparison = method1.toString().compareTo(method2.toString());
+			}
+		}
+		return comparison;
+	}
```

Comment:
Well, the Stream-like implementation results in 4 lambda expressions and 2 method references; whereas, the original implementation uses simple nested if-statements.

So I can only assume that the original implementation performs better (even if only marginally).

Consequently, since looking up methods is an often used, core, low-level feature of the JUnit Platform, I would vote for keeping the original implementation _as is_.

The readability or clarity of such an implementation does not play a big role here, especially since typically only core committers would ever even look at that private utility method.